### PR TITLE
Make mdx compatible with ocaml 5.2

### LINKED ISSUE #447: OCaml 5.2 support
Currently mdx does not compile with OCaml 5.2:
```
#=== ERROR while compiling mdx.2.3.1 ==========================================#
# context              2.2.0~beta2~dev | linux/x86_64 | ocaml-variants.5.2.0+trunk | file:///home/opam/opam-repository
# path                 ~/.opam/5.2/.opam-switch/build/mdx.2.3.1
# command              ~/.opam/5.2/bin/dune build -p mdx -j 1 @install
# exit-code            1
# env-file             ~/.opam/log/mdx-3916-cad682.env
# output-file          ~/.opam/log/mdx-3916-cad682.out
### output ###
# (cd _build/default && /home/opam/.opam/5.2/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/top/.mdx_top.objs/byte -I /home/opam/.opam/5.2/lib/astring -I /home/opam/.opam/5.2/lib/camlp-streams -I /home/opam/.opam/5.2/lib/csexp -I /home/opam/.opam/5.2/lib/findlib -I /home/opam/.opam/5.2/lib/fmt -I /home/opam/.opam/5.2/lib/logs -I /home/opam/.opam/5.2/lib/ocaml-version -I /home/opam/.opam/5.2/lib/ocaml/compiler-libs -I /home/opam/.opam/5.2/lib/ocaml/str -I /home/opam/.opam/5.2/lib/ocaml/threads -I /home/opam/.opam/5.2/lib/ocaml/unix -I /home/opam/.opam/5.2/lib/re -I /home/opam/.opam/5.2/lib/result -I /home/opam/.opam/5.2/lib/seq -I lib/.mdx.objs/byte -I vendor/odoc-parser/src/.odoc_parser.objs/byte -intf-suffix .ml -no-alias-deps -open Mdx_top__ -o lib/top/.mdx_top.objs/byte/mdx_top.cmo -c -impl lib/top/mdx_top.pp.ml)
# File "lib/top/mdx_top.ml", line 205, characters 29-37:
# Error: Unbound value "Exp.fun_"
```

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #447

### CODE REVIEW

File: test/bin/mdx-test/expect/ocaml-errors/test-case.md [hunk 1]
```diff
@@ -37,14 +37,14 @@ end
 
 Existing error blocks attached to a valid ocaml block are removed:
 
-```ocaml
+```ocaml version>=5.2
```
Comment (Julow) [inline test/bin/mdx-test/expect/ocaml-errors/test-case.md:40]: This block shouldn't be guarded, the error message is actually removed and not checked.
```suggestion
```ocaml
```

File: test/bin/mdx-test/expect/ocaml-errors-ellipsis/test-case.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+```ocaml version>=5.2
 module Counter: Irmin.Contents.S with type t = int64 = struct
   type t = int64
   let t = Irmin.Type.int64
 ```
 ```mdx-error
 ...
-Error: Syntax error: 'end' expected
+Error: Syntax error: end expected
 ...
```
Comment (Julow) [inline test/bin/mdx-test/expect/ocaml-errors-ellipsis/test-case.md:11]: This test finds a new bug: A empty line is added after the last ellipsis when the block is disabled due to version.

Comment (gpetiot) [reply test/bin/mdx-test/expect/ocaml-errors-ellipsis/test-case.md:11]: I don't have this bug locally. We can open another issue for later, but it shouldn't block this PR.

Comment (Julow) [reply test/bin/mdx-test/expect/ocaml-errors-ellipsis/test-case.md:11]: This bug is reproducible with ocaml `< 5.2` and is new.

Comment (gpetiot) [reply test/bin/mdx-test/expect/ocaml-errors-ellipsis/test-case.md]: This is now fixed in `lib/block.ml` (but I squashed everything).
Let me know what you think, but to me it looks ready to be merged.

**Review by Julow — APPROVED** [review]
Looks good :)

**Comment by kit-ty-kate** [comment]
Could we get a release by any chance?

**Pull request merged**